src/blenderbim/blenderbim/bim/module/georeference/operator.py

```diff
@@ -49,14 +49,7 @@ def execute(self, context):
             new.is_null = Data.projected_crs[attribute.name()] is None
             new.is_optional = attribute.optional()
             new.data_type = data_type
-            if data_type == "string":
-                new.string_value = "" if new.is_null else Data.projected_crs[attribute.name()]
-            elif data_type == "float":
-                new.float_value = 0.0 if new.is_null else Data.projected_crs[attribute.name()]
-            elif data_type == "integer":
-                new.int_value = 0 if new.is_null else Data.projected_crs[attribute.name()]
-            elif data_type == "boolean":
-                new.bool_value = False if new.is_null else Data.projected_crs[attribute.name()]
+            new.set_value(new.get_value_default() if new.is_null else Data.projected_crs[attribute.name()])
```

Comment:
I'd rather revert this change and rewrite it using the `import_attributes` helper. However, you will need a callback to handle the units. In addition, the units code needs to be rewritten to pull from available named units (now that we have a new units system) instead of the hardcoded bunch that we have right now. Would you be interested in tackling this? It would be a good intro to the IFC data side of things.

Reply:
Oh right `import_attributes` makes sense ! I understand why the callback is needed in other instances but here it looks like it isn't needed ?

Yeah I'd be fine tackling the unit linking refactor, feel free to point directions and I'll dig around :)